src/write_gcm.c

```diff
@@ -3244,26 +3335,30 @@ static int wg_json_CreateTimeSeries(
     wg_json_string(jc, "metric");
     wg_json_Metric(jc, head);
 
+    char *metric_value_type_string = wg_metric_value_type_to_string(
+      wg_metric_value_type_from_wg_payload_t(head));
+
     switch (head->values[0].ds_type) {
       case DS_TYPE_GAUGE:
-      wg_json_string(jc, "metricKind");
-      wg_json_string(jc, "GAUGE");
-      wg_json_string(jc, "valueType");
-      wg_json_string(jc, "DOUBLE");
-      break;
+        wg_json_string(jc, "metricKind");
+        wg_json_string(jc, "GAUGE");
+        wg_json_string(jc, "valueType");
+        wg_json_string(jc, metric_value_type_string);
+        break;
 
       case DS_TYPE_DERIVE:
       case DS_TYPE_COUNTER:
-      wg_json_string(jc, "metricKind");
-      wg_json_string(jc, "CUMULATIVE");
-      wg_json_string(jc, "valueType");
-      wg_json_string(jc, "INT64");
-      break;
+        wg_json_string(jc, "metricKind");
+        wg_json_string(jc, "CUMULATIVE");
+        wg_json_string(jc, "valueType");
+        wg_json_string(jc, metric_value_type_string);
+        break;
     }
 
     wg_json_string(jc, "points");
     wg_json_Points(jc, head);
 
+    sfree(metric_value_type_string);
```

Comment:
This is pointing at static strings, and shouldn't be freed.

Reply:
Done.